crates/core/src/kernel/snapshot/parse.rs

```diff
@@ -28,19 +32,20 @@ pub(super) fn read_metadata(batch: &dyn ProvidesColumnByName) -> DeltaResult<Opt
 
         for idx in 0..arr.len() {
             if arr.is_valid(idx) {
-                return Ok(Some(Metadata {
-                    id: ex::read_str(id, idx)?.to_string(),
-                    name: ex::read_str_opt(name, idx).map(|s| s.to_string()),
-                    description: ex::read_str_opt(description, idx).map(|s| s.to_string()),
-                    format: Default::default(),
-                    schema_string: ex::read_str(schema_string, idx)?.to_string(),
-                    partition_columns: collect_string_list(&partition_columns, idx)
+                let value = json!({
+                    "id": ex::read_str(id, idx)?.to_string(),
+                    "name": ex::read_str_opt(name, idx).map(|s| s.to_string()),
+                    "description": ex::read_str_opt(description, idx).map(|s| s.to_string()),
+                    "format": { "provider": "parquet", "options": {} },
+                    "schemaString": ex::read_str(schema_string, idx)?.to_string(),
+                    "partitionColumns": collect_string_list(&partition_columns, idx)
```

Comment:
As I'm reading this, let's say we _have_ to keep this `json!` hack in place, could you perhaps leave a comment at each site with a link to a github issue to remove it? That way we can come back along later and find all the invocations to clean up. I worry about missing a spot in the future and us having to chase pretty tricky bugs

Reply:
opened https://github.com/delta-io/delta-kernel-rs/issues/1055 to terack sthis and sprkinkled comments in the codebase.